### ISSUE TITLE
Wrong lifted IL to LLIL conversion

### ISSUE THREAD

**Issue by skr0x1c0** [Vector35/binaryninja-api#5806]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5747
- OS: macOS 15.0 Beta
- OS Version: 24A5298h
- CPU Architecture: M1

**Bug Description:**
In several occasions common in aarch64 binaries, conversion from lifted IL to LLIL is wrong.
##### Disassembly:
``` asm
00000000  004001b1   adds    x0, x0, #0x50
00000004  e0379f1a   cset    w0, cs
00000008  c0035fd6   ret     
```
##### Lifted IL:
``` asm
0 @ 00000000  x0 = add.q{*}(x0, 0x50)
1 @ 00000004  w0 = cs {int.u>=, float.f>=} ? 1 : 0
2 @ 00000008  <return> jump(x30)
```
##### LLIL:
``` asm
0 @ 00000000  temp0.q = x0
1 @ 00000000  x0 = x0 + 0x50
2 @ 00000000  flag:c = temp0.q u>= -0x50
3 @ 00000004  w0 = not(flag:c) ? 1 : 0 ; Should be w0 = flag:c ? 1 : 0
4 @ 00000008  <return> jump(x30)
```

##### Disassembly:
``` asm
00000018  000001ab   adds    x0, x0, x1
0000001c  42000054   b.hs    0x24
00000020  a00080d2   mov     x0, #0x5
00000024  c0035fd6   ret     
```
##### Lifted IL:
``` asm
0 @ 00000018  x0 = add.q{*}(x0, x1)
1 @ 0000001c  if (cs {int.u>=, float.f>=}) then 4 @ 0x24 else 6 @ 0x20
4 @ 00000024  <return> jump(x30)
6 @ 00000020  x0 = 5
7 @ 00000020  goto 4 @ 0x24
```
##### LLIL:
``` asm
0 @ 00000018  temp0.q = x0
1 @ 00000018  x0 = x0 + x1
2 @ 00000018  flag:c = temp0.q + x1 u< temp0.q
3 @ 0000001c  if (not(flag:c)) then 4 @ 0x24 else 5 @ 0x20 ; Should be if(flag:c) then 4 @ 0x24 else 5 @ 0x20
4 @ 00000024  <return> jump(x30)
5 @ 00000020  x0 = 5
6 @ 00000020  goto 4 @ 0x24
```

**Steps To Reproduce:**
See 1st and 3rd functions in attached binary 
[test.zip](https://github.com/user-attachments/files/16419493/test.zip)

Please fix this bug on priority because this makes Binary Ninja unusable with almost all Apple aarch64 binaries.

**Comment by galenbwill** [comment]
It does appear that something weird is happening between Lifted IL and LLIL, and this goes back to at least 4.0 stable (though I suspect farther back than that). It does not actually appear to be an Aarch64 issue, but something happening in the core -- however, I haven't been able to verify that. Someone else will need to look at this too, so I'm putting "Awaiting Triage" back on it.

**Comment by D0ntPanic** [comment]
Fixed in 4.2.5840
